pkg/endpoint/bpf.go

```diff
@@ -1174,6 +1227,59 @@ func (e *Endpoint) applyPolicyMapChanges() error {
 	//  applied to the Endpoint's bpf policy map.
 	closer, changes := e.desiredPolicy.ConsumeMapChanges()
 	defer closer()
+	var deleteLen int
+	for k := range changes.Deletes {
+		if _, ok := changes.Adds[k]; !ok {
+			deleteLen++
+		}
+	}
+	// If the desiredPolicy will be larger than the BPF maximum after
+	// the changes we need to reset the consumed changes and either
+	// lockdown or return an inevitable error (which would happen on
+	// insertion if we proceed).
+	if e.desiredPolicy.GetPolicyMap().Len()+len(changes.Adds)-deleteLen > policymap.MaxEntries {
+		// Revert the map changes that were just consumed.
+		e.desiredPolicy.RevertChanges(changes)
+
+		// Avoid an insertion error further down and return an
+		// error now.
+		if !option.Config.EnableEndpointLockdownOnPolicyOverflow {
+			// Do not return ErrPolicyEntryMaxExceeded as it
+			// implies that option.Config.EnableEndpointLockdownOnPolicyOverflow
+			// is "true".
+			return fmt.Errorf("policy map exceeds the map max entries limit")
+		}
+
+		e.getLogger().WithFields(logrus.Fields{
+			logfields.EndpointID: e.ID,
+		}).Warnf("The policy map exceeds the max entries limit, %s is enabled, locking the endpoint down.",
+			option.EnableEndpointLockdownOnPolicyOverflow)
```

Comment:
Would it be possible to add metrics to add a method for tracking when an endpoint is locked that doesn't require watching logs?

Reply:
We shouldn't do this. Policy map pressure can already provide the same information. I'll grant that it seems like "something" has occurred, but that is just that map pressure has hit "1.0". A metric behind a config flag can cause problems as well and is another point against creating a metric for this. I will make sure to add documentation to the section on the flag to encourage good map pressure monitoring

Reply:
Sounds good, thanks!